helm/argocd-test/action.yml

```
@@ -13,28 +13,68 @@ inputs:
   github-token:
     description: Specify the github token to login to ghcr.io
     required: true
-
+  helm-version:
+    description:
+    required: true
+    default: "v3.13.3"
+  kubernetes-tested-version:
+    description: Specify the kubernetes version used to test helm templates
+    required: true
+    default: "v1.27.9"
+  kuberconform-version:
+    description: Specify the kuberconform version.
+    required: true
+    default: "v0.6.3"
+  datatree-crd-catalog-version:
+    description: Specify the CRD catalog version .
+    required: true
+    default: "v0.0.12"
 
 runs:
   using: composite
   steps:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Set up Helm
-        uses: azure/setup-helm@v3
-        with:
-          version: "v3.13.3"
-
       - name: Login to ghcr.io
         uses: docker/login-action@v3
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ inputs.github-token }}
 
+      - name: Set up Helm
+        uses: azure/setup-helm@v3
+        with:
+          version: ${{ inputs.helm-version }}
+
       - name: Run helm template
         shell: bash
+        env:
+          HELM_CHART_FOLDER: ${{ inputs.chart-dir }}
+          HELM_OPTIONS: ${{ inputs.helm-options }}
+        run: |
+          set -e
+          HERE=$(pwd)
+          cd $HELM_CHART_FOLDER
+          helm dependency build .
+          helm template . $HELM_OPTIONS > helm-rendered.yaml
+          mv helm-rendered.yaml ${HERE}/
+
+      - name: Set up kubeconform
+        shell: bash
+        env:
+          KUBECONFORM_VERSION: ${{ inputs.kuberconform-version }}
+          KUBECONFORM_BASE_URL: "https://github.com/yannh/kubeconform/releases/download"
+        run: |
+          set -o pipefail
+          OS=$(uname)
+          curl -sSfL "${KUBECONFORM_BASE_URL}/${KUBECONFORM_VERSION}/kubeconform-${OS}-amd64.tar.gz" | tar -xzf - kubeconform
+
+      - name: Run kubeconform
+        shell: bash
+        env:
+          KUBERNETES_VERSION: ${{ inputs.kubernetes-tested-version}}
+          DATATREE_CRD_CATALOG_URL: 'https://raw.githubusercontent.com/datreeio/CRDs-catalog/${{ inputs.datatree-crd-catalog-version }}/{{.Group}}/{{.ResourceKind}}_{{.ResourceAPIVersion}}.json'
         run: |
-          cd ${{ inputs.chart-dir }}
-          helm dependency build . && helm template . ${{ inputs.helm-options }}
+          ./kubeconform -strict -exit-on-error -summary -schema-location default -schema-location $DATATREE_CRD_CATALOG_URL helm-rendered.yaml
```

You could do this:

```suggestion
          ./kubeconform -strict -exit-on-error -summary -schema-location default -schema-location $DATATREE_CRD_CATALOG_URL $HELM_CHART_FOLDER/helm-rendered.yaml
```

And remove the logic with the `$HERE` var I believe